### PR TITLE
Mask-rcnn std case

### PR DESCRIPTION
**Pull request by yuzhou03**

### CODE REVIEW

**Comment by yuzhou03** [comment]
evaluater.py
utils/* 
dataloader.py， 除了最上面四个方法（build_train_dataset， build_eval_dataset， build_train_dataloader， build_eval_dataloader）的所有代码均为copy from tv，未加修改

File: training/benchmarks/mask_rcnn/pytorch/config/_base.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+warmup = 0.1
+
+# =========================================================
+# utils
+# =========================================================
+seed: int = 0
+dist_backend: str = 'nccl'
+num_workers: int = 4
+device: str = None
+gpu: int = None
```
Comment (shh2000) [inline training/benchmarks/mask_rcnn/pytorch/config/_base.py:57]: remove gpu

File: training/benchmarks/mask_rcnn/pytorch/run_pretraining.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    start = time.time()
+    config_update, state = main()
+    if not dist_pytorch.is_main_process():
+        sys.exit(0)
+
+    # 训练信息写日志
+    e2e_time = time.time() - start
+
+    if config_update.do_train:
+        training_perf = state.num_trained_samples * config_update.n_device / state.raw_train_time
```
Comment (shh2000) [inline training/benchmarks/mask_rcnn/pytorch/run_pretraining.py:129]: 不需要乘。可以仔细看一下处理num_trained_samples的逻辑。里面加的是len(dataset)而不是len(loader)

File: training/benchmarks/mask_rcnn/pytorch/config/_base.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
+# DO NOT MODIFY THESE REQUIRED PARAMETERS
+
+# Required parameters
+vendor: str = "nvidia"
+data_dir: str = None
+name: str = "retinanet"
```
Comment (shh2000) [inline training/benchmarks/mask_rcnn/pytorch/config/_base.py:6]: name

Comment (shh2000) [reply training/benchmarks/mask_rcnn/pytorch/config/_base.py:6]: 这里写错的话，能正确跑出log吗

File: training/nvidia/mask_rcnn-pytorch/README.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+   - 加速卡驱动版本：470.129.06
+   - Docker 版本：20.10.16
+   - 训练框架版本：pytorch-1.8.0a0+52ea372
+   - 依赖软件版本：无
+
+
+### 运行情况
+| 训练资源 | 配置文件        | 运行时长(s) | 目标mAP精度(bbox && segm) | 收敛mAP精度(bbox && segm) | 性能(samples/s) |
+| -------- | --------------- | ----------- | ------------------------- | ------------------------- | --------------- |
+| 单机8卡  | config_A100x1x8 | 14108.69    | 0.3558 && 0.3185          | 0.3751 && 0.3386          | 1136.00         |
```
Comment (shh2000) [inline training/nvidia/mask_rcnn-pytorch/README.md:25]: 这个性能可能要重新算一下，或者直接手动/8

File: training/benchmarks/mask_rcnn/pytorch/config/_base.py (hunk 3 above)
Comment (yuzhou03) [reply training/benchmarks/mask_rcnn/pytorch/config/_base.py:6]: done

**Comment by yuzhou03** [comment]
A100 1x8 log
[faster_rcnn_nv_1x8.zip](https://github.com/FlagOpen/FlagPerf/files/11634437/faster_rcnn_nv_1x8.zip)

File: training/nvidia/mask_rcnn-pytorch/README.md (hunk 4 above)
Comment (yuzhou03) [reply training/nvidia/mask_rcnn-pytorch/README.md]: done

File: training/benchmarks/mask_rcnn/pytorch/run_pretraining.py (hunk 2 above)
Comment (yuzhou03) [reply training/benchmarks/mask_rcnn/pytorch/run_pretraining.py]: done

File: training/benchmarks/mask_rcnn/pytorch/config/mutable_params.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+mutable_params = [
+    'vendor', 'data_dir', 'train_data', 'eval_data', 'lr', 'weight_decay',
+    'momentum', 'lr_steps', 'lr_gamma', 'train_batch_size', 'eval_batch_size',
+    'do_train', 'fp16', 'distributed', 'warmup', 'dist_backend', 'num_workers',
+    'device', 'gpu'
```
Comment (shh2000) [inline training/benchmarks/mask_rcnn/pytorch/config/mutable_params.py:5]: 没修干净

File: training/nvidia/mask_rcnn-pytorch/README.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+   - 加速卡驱动版本：470.129.06
+   - Docker 版本：20.10.16
+   - 训练框架版本：pytorch-1.8.0a0+52ea372
+   - 依赖软件版本：无
+
+
+### 运行情况
+| 训练资源 | 配置文件        | 运行时长(s) | 目标mAP精度(bbox && segm) | 收敛mAP精度(bbox && segm) | 性能(samples/s) |
+| -------- | --------------- | ----------- | ------------------------- | ------------------------- | --------------- |
+| 单机8卡  | config_A100x1x8 | 14108.69    | 0.3558 && 0.3185          | 0.3751 && 0.3386          | 142.0           |
```
Comment (shh2000) [inline training/nvidia/mask_rcnn-pytorch/README.md:25]: 既然用16/0.16跑的，target设置的这么低吗？这个14108.69s，是跑到0.3751，还是跑到0.3558出来的时间？

File: training/benchmarks/mask_rcnn/pytorch/config/mutable_params.py (hunk 5 above)
Comment (yuzhou03) [reply training/benchmarks/mask_rcnn/pytorch/config/mutable_params.py]: remove gpu from mutable_params.py

File: training/nvidia/mask_rcnn-pytorch/README.md (hunk 6 above)
Comment (yuzhou03) [reply training/nvidia/mask_rcnn-pytorch/README.md:25]: 1. 之前那个PR，bs=16/lr=0.12就是这个target.
2. 是第17个epoch，跑到实际收敛精度的时间。

File: training/benchmarks/mask_rcnn/README.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+```
+
+
+
+### 模型checkpoint 
+预训练权重下载地址（下载后放入data_dir文件夹中）
+
+#### Resnet50预训练权重 
+ https://download.pytorch.org/models/resnet50-0676ba61.pth 
+ (注意，下载预训练权重后要重命名， 比如在train.py中读取的是resnet50.pth文件，不是resnet50-0676ba61.pth)
```
Comment (shh2000) [inline training/benchmarks/mask_rcnn/README.md:50]: 这句话是啥意思

Comment (shh2000) [reply training/benchmarks/mask_rcnn/README.md:50]: 还要查一下不是copy from retina的部分，比如两个文档。有没有wzmiaomiao的遗留问题

Comment (yuzhou03) [reply training/benchmarks/mask_rcnn/README.md:50]: mask-rcnn的create_model中用的是本地文件，url_resnet50_backbone = "file://" + resnet50_pretrained_path。
因此这里无需修改。

**Review by shh2000 — APPROVED** [review]

**Review by upvenly — APPROVED** [review]
